test/RpcCompatibility.js

```diff
@@ -252,7 +252,19 @@ describe('RpcCompatibility', () => {
 
         expect(estimatedGas).to.be.a('string');
         expect(gas).to.be.gt(0);
-        expect(gas).to.be.lt(30000); // report if gas estimation is different than ethereum
+        expect(gas).to.be.lt(30000); // report if gas estimation does not matches Ethereum's behavior to adjust core configuration if necessary.
+    });
+
+    it('should send tx with estimated gas', async () => {
+        const newValue = 300;
+        const txParams = {
+            to: rpcCompatibilityContract.address,
+            data: rpcCompatibilityContract.interface.encodeFunctionData('updateValue', [newValue]),
+        };
+
+        const estimatedGas = await provider.estimateGas(txParams);
+        const receipt = await rpcCompatibilityContract.updateValue(newValue, { gasLimit: estimatedGas }).then((tx) => tx.wait());
```

Comment:
when updateValue is called without gas limit, ethers already estimates the gas needed so this should already be covered. but i'm fine with being explicit in a test.